src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -220,72 +168,75 @@ private GapicSurfaceTestAssertView createAssertView(String expected, String actu
-    for (Field field : fields) {
-      String initString = prefix + field.getSimpleName();
-
-      if (field.getType().isMap()) {
-        // TODO(michaelbausor) support recursive map initialization
-        fieldStrings.add(initString);
-      } else {
-        if (field.getType().isRepeated()) {
-          initString += "[0]";
-        }
-        fieldStrings.add(initString);
-        if (field.getType().isMessage() && depth > 0) {
-          createTestRequestInitFieldStructure(
-              context,
-              fieldStrings,
-              field.getType().getMessageType().getFields(),
-              initString + ".",
-              depth - 1);
-        }
-      }
-    }
+      Iterable<InitCodeNode> orderedItems,
+      Iterable<InitCodeNode> argItems) {
+    ImportTypeTransformer importTypeTransformer = new ImportTypeTransformer();
+    ModelTypeTable typeTable = context.getTypeTable();
+    SurfaceNamer namer = context.getNamer();
+
+    // Initialize the type table with the apiClassName since each sample will be using the
+    // apiClass.
+    typeTable.getAndSaveNicknameFor(
+        namer.getFullyQualifiedApiWrapperClassName(
+            context.getInterface(), context.getApiConfig().getPackageName()));
+
+    return InitCodeView.newBuilder()
+        .lines(generateSurfaceInitCodeLines(context, orderedItems))
+        .fieldSettings(getFieldSettings(context, argItems))
+        .imports(importTypeTransformer.generateImports(typeTable.getImports()))
+        .apiFileName(
+            namer.getServiceFileName(
+                context.getInterface(), context.getApiConfig().getPackageName()))
+        .build();
   }
 
-  private Map<String, Object> createMockResponseInitFieldStructure(
-      MethodTransformerContext context) {
-    ArrayList<String> fields = new ArrayList<>();
-    for (Field field : context.getMethod().getOutputMessage().getFields()) {
-      if (field.getType().isPrimitive()) {
-        fields.add(field.getSimpleName());
-      }
-    }
+  private List<InitCodeNode> createMockResponseInitSubTrees(MethodTransformerContext context) {
+    List<InitCodeNode> specItems = new ArrayList<>();
     if (context.getMethodConfig().isPageStreaming()) {
-      // Initialize one resource element if it is page-streaming, and set the initial value of the
-      // page token to empty, in order to indicate that no more pages are available.
+      // Initialize one resource element if it is page-streaming.
       PageStreamingConfig config = context.getMethodConfig().getPageStreaming();
-      fields.add(config.getResourcesField().getSimpleName() + "[0]");
-      fields.add(config.getResponseTokenField().getSimpleName() + "=\"\"");
+      String resourceFieldName = config.getResourcesField().getSimpleName();
+      InitCodeNode childItem = InitCodeNode.create("0");
+      InitCodeNode item =
+          InitCodeNode.createWithChildren(
+              resourceFieldName, InitCodeLineType.ListInitLine, childItem);
```

Comment:
Maybe for readability here, support this:

InitCodeNode item = InitCodeNode.createSingletonList(resourceFieldName);

which creates a InitCodeLineType.ListInitLine with a child item "0". 

And also use it for the bundling case below.